File: app/src/main/java/com/callmangement/EHR/api/APIClient.java
package com.callmangement.EHR.api;

import android.content.Context;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class APIClient {

    public static Retrofit GetRetrofitClientWithoutHeaders(final Context context, final String BASE_URL) {



        // Create logging interceptor to debug request and response
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);

        // Create an interceptor to add headers to every request
        Interceptor headerInterceptor = new Interceptor() {
            @Override
            public Response intercept(Chain chain) throws IOException {
                Request originalRequest = chain.request();
                Request.Builder requestBuilder = originalRequest.newBuilder()
                        .header("Content-Type", "application/json; charset=utf-8");
                Request request = requestBuilder.build();
                return chain.proceed(request);
            }
        };

   OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
        httpClient.readTimeout(Constants.CONNECTION_TIME_OUT, TimeUnit.SECONDS);
        httpClient.connectTimeout(Constants.CONNECTION_TIME_OUT, TimeUnit.SECONDS);
        httpClient.addInterceptor(headerInterceptor);
        httpClient.addInterceptor(logging);
        OkHttpClient client = httpClient.build();
        return new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build();
    }












}

File: app/src/main/java/com/callmangement/ehr/api/APIClient.kt
package com.callmangement.ehr.api

import android.content.Context
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

object APIClient {

    fun getRetrofitClientWithoutHeaders(context: Context, BASE_URL: String): Retrofit {
        val logging = HttpLoggingInterceptor().apply {
            level = HttpLoggingInterceptor.Level.BODY
        }

        val headerInterceptor = Interceptor { chain ->
            val originalRequest = chain.request()
            val request = originalRequest.newBuilder()
                .header("Content-Type", "application/json; charset=utf-8")
                .build()
            chain.proceed(request)
        }

        val httpClient = OkHttpClient.Builder()
            .readTimeout(Constants.CONNECTION_TIME_OUT?.toLong()!!, TimeUnit.SECONDS)
            .connectTimeout(Constants.CONNECTION_TIME_OUT?.toLong()!!, TimeUnit.SECONDS)
            .addInterceptor(headerInterceptor)
            .addInterceptor(logging)
            .build()

        return Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .client(httpClient)
            .build()
    }
}